src/components/app.tsx

```diff
@@ -5,13 +5,41 @@ import { DragDropContext } from 'react-dnd';
 import HTML5Backend from 'react-dnd-html5-backend';
 import * as SplitPane from 'react-split-pane';
 
+import { Dispatch } from 'redux';
+import { StateWithHistory } from 'redux-undo';
+import {Data, InlineData, isInlineData, isUrlData, UrlData} from 'vega-lite/build/src/data';
+import { datasetUrlLoad } from '../actions';
+import { StateBase } from '../models/index';
+
 import {DataPane} from './data-pane';
 import {EncodingPane} from './encoding-pane';
 import {Header} from './header';
 import {ViewPane} from './view-pane';
 
 
-class AppBase extends React.PureComponent<any, {}> {
+export type VoyagerConfig = Object;
+export type VoyagerData = Data;
+
+interface Props extends React.Props<AppBase> {
+  config?: VoyagerConfig;
+  data?: Data;
+  dispatch: Dispatch<StateWithHistory<Readonly<StateBase>>>;
+}
+
+class AppBase extends React.PureComponent<Props, {}> {
+
+  constructor(props: any) {
```

Comment:
This needs to be `any`. If you change it to `Props` it will not compile. After much much searching and experimentation, I believe it is because of the HTML5Backend decorator. See http://stackoverflow.com/questions/40269415/typescript-2-0-with-react-dnd-gives-error-jsx-element-attributes-may-not-be-a-u for more details. 

I also think this is what prevents this being compatible with connect. So I am passing dispatch in directly.

Though do feel free to take a look in case i've overlooked something.